evalml/pipelines/components/estimators/classifiers/decision_tree_classifier.py

```diff
@@ -15,13 +15,25 @@ class DecisionTreeClassifier(Estimator):
         "max_features": ["auto", "sqrt", "log2"],
         "max_depth": Integer(4, 10),
     }
+    """{
+        "criterion": ["gini", "entropy"],
+        "max_features": ["auto", "sqrt", "log2"],
+        "max_depth": Integer(4, 10),
+    }"""
     model_family = ModelFamily.DECISION_TREE
+    """ModelFamily.DECISION_TREE"""
     supported_problem_types = [
         ProblemTypes.BINARY,
         ProblemTypes.MULTICLASS,
         ProblemTypes.TIME_SERIES_BINARY,
         ProblemTypes.TIME_SERIES_MULTICLASS,
     ]
+    """[
+        ProblemTypes.BINARY,
+        ProblemTypes.MULTICLASS,
+        ProblemTypes.TIME_SERIES_BINARY,
+        ProblemTypes.TIME_SERIES_MULTICLASS,
+    ]"""
```

Comment:
This is really weird and I'm not sure I like it.  It adds a lot of bloat, it seems.  Probably still worth the performance benefit, but will take some getting used to.